Source/PropertyTools.Wpf/PropertyGrid/MetadataPropertyItemFactory.cs

```diff
@@ -0,0 +1,735 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="MetadataPropertyItemFactory.cs" company="PropertyTools">
+//   Copyright (c) 2014 PropertyTools contributors
+// </copyright>
+// <summary>
+//   Provides a property item factory for the PropertyGrid control that uses the MetadataAttribute.
+// </summary>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace PropertyTools.Wpf
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.ComponentModel;
+    using System.ComponentModel.DataAnnotations;
+    using System.Globalization;
+    using System.Linq;
+    using System.Windows;
+    using System.Windows.Data;
+
+    using PropertyTools.DataAnnotations;
+
+    /// <summary>
+    /// Provides a property item factory for the <see cref="PropertyGrid" /> control.
+    /// </summary>
+    public class MetadataPropertyItemFactory : IPropertyItemFactory
```

Comment:
isn't there a lot of code duplication here? This means it will be double amount of work to implement #11, I think. Can you base it on DefaultPropertyItemFactory? Or extract a new base class?